versioned/persist/inmem/src/main/java/org/projectnessie/versioned/persist/inmem/InmemoryDatabaseAdapter.java

```diff
@@ -252,4 +257,69 @@ protected RefLog fetchFromRefLog(NonTransactionalOperationContext ctx, Hash refL
         .map(ProtoSerialization::protoToRefLog)
         .collect(Collectors.toList());
   }
+
+  @Override
+  protected void writeAttachments(Stream<Entry<AttachmentKey, AttachmentValue>> attachments) {
+    attachments.forEach(
+        e -> store.attachments.put(dbKey(e.getKey().toByteString()), e.getValue().toByteString()));
+  }
+
+  @Override
+  protected boolean consistentWriteAttachment(
+      AttachmentKey key, AttachmentValue value, Optional<String> expectedVersion) {
+    try {
+      store.attachments.compute(
+          dbKey(key.toByteString()),
+          (k, current) -> {
+            if (expectedVersion.isPresent()) {
```

Comment:
Do we have guidelines around handling the `Optional` type?
can we use `expectedVersion.map().orElse()` instead of `if(isPresent())...`?

Reply:
We don't have guidelines - and I think we shouldn't have.
IMO it should be "readable". As long as the "conventional" approach is readable (or "equally readable" compared to stream-ish one, it's fine, because the implicit one's less work for the VM.

Reply:
I am asking because many OS projects have guidelines around handling Optional. A lot of them prefer the non-functional approach (`isPresent`) and some of them functional (`map()..`) so it maybe beneficial to have unified rules regarding `Optional` handling